pkg/kubelet/cm/devicemanager/topology.go

```diff
@@ -26,34 +25,36 @@ func (m *ManagerImpl) GetTopologyHints(pod v1.Pod, container v1.Container) ([]to
 		resource := string(resourceObj)
 		amount := int64(amountObj.Value())
 		if m.isDevicePluginResource(resource) {
-			klog.Infof("%v is a resource managed by device manager.", resource)
-			klog.Infof("Health Devices: %v", m.healthyDevices[resource])
+			klog.Infof("[devicemanager-topology] %v is a resource managed by device manager.", resource)
 			containerRequiresDevice = true
 			if _, ok := m.healthyDevices[resource]; !ok {
-				klog.Infof("No healthy devices for resource %v", resource)
+				klog.Infof("[devicemanager-topology] No healthy devices for resource %v", resource)
+				continue
+			}
+
+			if aligned := checkIfDeviceHasSocketAlignment(resource, m.allDevices); !aligned {
+				klog.Infof("[devicemanager-topology] Device does not have a topology preference")
 				continue
 			}
 			available := m.getAvailableDevices(resource)
 
 			if int64(available.Len()) < amount {
-				klog.Infof("requested number of devices unavailable for %s. Requested: %d, Available: %d", resource, amount, available.Len())
+				klog.Infof("[devicemanager-topology] Requested number of devices unavailable for %s. Requested: %d, Available: %d", resource, amount, available.Len())
 				continue
 			}
-			klog.Infof("[devicemanager] Available devices for resource %v: %v", resource, available)
+			klog.Infof("[devicemanager-topology] Available devices for resource %v: %v", resource, available)
 
 			deviceSocketAvail := getDevicesPerSocket(resource, available, m.allDevices)
 
 			for socket, amountAvail := range deviceSocketAvail {
-				klog.Infof("Socket: %v, Avail: %v AmountReq: %v", socket, amountAvail, amount)
 				mask, _ := socketmask.NewSocketMask(int(socket))
```

Comment:
I think this could go inside the conditional at line 51 since we don't need it if that conditional evaluates to "false"